elytron/WFCORE-5178_native-support-for-oidc.adoc

```diff
@@ -0,0 +1,178 @@
+additional work so it might be done as a follow-up task.
+
+* *Migration*: For the most part, the same subsystem attributes and JSON configuration options that could
+be used with the `keycloak-adapter-subsystem` and the `keycloak.json` file will still work with the
+`elytron-oidc-subsystem` and the `oidc.json` file. The main difference is the actual subsystem name and
+the name of the JSON configuration file. That means that the migration tool could potentially detect an
+existing `keycloak` subsystem configuration and then change its name to the new subsystem
+name instead (and update any attributes/elements if needed). For the deployment side, the migration tool
+could detect deployments with `auth-method` set to KEYCLOAK and with a keycloak.json configuration file and
+update the `auth-method` to OIDC and rename the configuration file to oidc.json.
+
+=== Non-Requirements
+
+This integration will not be supporting the use of pre-configured SecurityDomains. The purpose of OIDC
+is to allow a client to verify the identity of an end-user based on authentication that is performed
+by an authorization server. Information about the end-user can be obtained from ID tokens. Scope
+information (i.e., role information) can be obtained from access tokens. Thus, we do not need
+pre-configured security resources in the Elytron subsystem.
+
+Support for the following optional OIDC specifications are currently non-requirements. These
+are also not provided by the Keycloak Java client adapter:
+
+* https://openid.net/specs/openid-connect-discovery-1_0.html[Discovery]
+* https://openid.net/specs/openid-connect-registration-1_0.html[Dynamic Registration]
+* https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html[OAuth 2.0 Multiple Response Types]
+* https://openid.net/specs/oauth-v2-form-post-response-mode-1_0.html[OAuth 2.0 Form Post Response Mode]
+* https://openid.net/specs/openid-connect-session-1_0.html[Session Management]
+* https://openid.net/specs/openid-connect-frontchannel-1_0.html[Front-Channel Logout]
+* https://openid.net/specs/openid-connect-backchannel-1_0.html[Back-Channel Logout]
+* https://openid.net/specs/openid-connect-federation-1_0.html[OpenID Connect Federation]
+
+== Test Plan
+
+This will require some discussion with QE to determine the best approach for testing. We will
+need to decide if it is acceptable to add dependencies on Keycloak (or another OIDC provider)
+to our testsuites or if we need to mock an OIDC provider.
+
+== Community Documentation
+
+A new section will be added to the Elytron documentation that describes:
+
+ * How to activate the OIDC integration for a deployment
+ * The new `elytron-oidc-client` subsystem and how to configure it
+ * The OIDC configuration properties that can be specified via a JSON file or via subsystem configuration
+ * How to use the new Galleon `oidc-client` layer
+
+It would be good to have a quickstart for this. This can be handled under its own RFE.
```

Comment:
@fjuma Can you create and add the new qs RFE please?

Reply:
Sure, will just discuss this one on our team call first tomorrow and then will create it.